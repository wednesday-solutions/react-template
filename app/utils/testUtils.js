import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter, Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import { IntlGlobalProvider } from '@components/IntlGlobalProvider';
import { theme } from '@containers/App';

export const renderWithIntl = (children) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );

export const renderProvider = (children, history) => {
  const store = configureStore({}, browserHistory).store;
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider theme={theme}>
          {history ? <Router history={history}>{children}</Router> : <BrowserRouter>{children}</BrowserRouter>}
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const getStyles = () => {};
