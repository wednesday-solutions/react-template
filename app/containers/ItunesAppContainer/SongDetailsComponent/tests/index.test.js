/**
 *
 * Tests for SongDetailsComponent
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import { SongDetailsComponent } from '../index';

describe('<SongDetailsComponent />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<SongDetailsComponent />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 SongDetailsComponent component', () => {
    const { getAllByTestId } = renderWithIntl(<SongDetailsComponent />);
    expect(getAllByTestId('song-details-component').length).toBe(1);
  });
});
