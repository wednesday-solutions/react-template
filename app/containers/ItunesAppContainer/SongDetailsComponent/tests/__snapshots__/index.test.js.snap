// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongDetailsComponent /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      data-testid="song-details-component"
    >
      <div
        style="display: flex;"
      >
        <div
          style="flex: 1; margin: 20px;"
        >
          <div
            class="SongThumbNail__StyledImage-vw8usd-2 daTdfv"
            height="280"
          >
            <div
              class="SongThumbNail__ThumbNailContainer-vw8usd-3 faiNZh"
            >
              <div>
                <div
                  style="flex: 1;"
                />
                <span
                  aria-label="pause-circle"
                  class="anticon anticon-pause-circle SongThumbNail__CustomPauseCircleOutlined-vw8usd-1 ikvBxO"
                  data-testid="pause_btn"
                  role="img"
                  tabindex="-1"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="pause-circle"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm-88-532h-48c-4.4 0-8 3.6-8 8v304c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V360c0-4.4-3.6-8-8-8zm224 0h-48c-4.4 0-8 3.6-8 8v304c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V360c0-4.4-3.6-8-8-8z"
                    />
                  </svg>
                </span>
                <div
                  style="flex: 1;"
                />
              </div>
            </div>
            <audio>
              <source />
            </audio>
          </div>
        </div>
        <div
          class="SongDetailsComponent__Container-sc-1y90cvw-0 XognC"
        >
          <span
            style="font-size: 20px;"
          />
          <span
            style="font-size: 10px;"
          />
          <div
            class="SongDetailsComponent__ArtistName-sc-1y90cvw-1 dBZBEX artistName"
          />
          <span
            style="text-transform: uppercase;"
          >
            :
            <span
              style="margin-left: 10px;"
            >
              NaN
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
